src/analysis/normed_space/basic.lean

```diff
@@ -94,6 +94,14 @@ variables [semi_normed_ring α]
 lemma norm_mul_le (a b : α) : (∥a*b∥) ≤ (∥a∥) * (∥b∥) :=
 semi_normed_ring.norm_mul _ _
 
+lemma nnnorm_mul_le (a b : α) : ∥a * b∥₊ ≤ ∥a∥₊ * ∥b∥₊ :=
+by simpa only [←norm_to_nnreal, ←real.to_nnreal_mul (norm_nonneg _)]
+  using real.to_nnreal_mono (norm_mul_le _ _)
+
+lemma nnnorm_pow_succ_le (a : α) (n : ℕ) : ∥a ^ n.succ∥₊ ≤ ∥a∥₊ ^ n.succ :=
```

Comment:
Isn't this true for all `n`? Also, does the `norm` version exist? If not, please provide that as well.

Reply:
No, it's not true for all `n` without `[norm_one_class A]` because a semi-normed ring doesn't necessarily force `∥1∥₊ = 1`. There is `norm_pow_le'` and `norm_pow_le` for the norm. I'll update the `nnnorm` versions to mirror these (and include the one for all `n` with the type class constraint).

Reply:
I mirrored the statements of the `norm` version, but then I just simplified everything, including the proofs of the `norm` versions by using the `nnnorm` versions.